src/harness/testcases/WINNF_FT_S_PCR_testcase.py

```diff
@@ -179,22 +172,93 @@ def triggerCpasAndRetrievePpaZone(self, ppa_id, ssl_cert, ssl_key):
-    except KeyError:
-      self.fail('There is an error while filtering PPA Zone record based '
-                'on PPA ID:{0}'.format(ppa_id))
-    else:
-      return uut_ppa_zone_data[0]
+    """
+
+    for device in registration_request:
+      if 'installationParam' not in device:
+        for conditional_params in conditional_registration_data:
+          # Check if FCC_ID+Serial_Number present in registrationRequest
+          # and conditional_params match and add the 'installationParam'.
+          self.assertIn('installationParam', conditional_params,
+                        msg='installationParm Object is not found in REG-Conditionals')
+          if not (conditional_params['fccId'] == device['fccId'] and \
+                  conditional_params['cbsdSerialNumber'] == device['cbsdSerialNumber']):
+            raise Exception('ConfigError:Wrong REG-Conditional data for device is found. '
+                            'Please load the correct REG-Conditional data for the device')
+          else:
+            # The following REG-conditional parameters are required to present in
+            # installationParam Object and cbsdCategory in RegistrationRequest for PPA
+            # reference model to determine PPA contour boundary.
+            # assert that all the needed parameters for PPA are present.
+            if any([conditional_param_name not in conditional_params['installationParam']
+                    for conditional_param_name in CONDITIONAL_PARAMS_REQUIRED]):
+              raise Exception('ConfigError:Any one of the REG conditional parameter:%s '
+                              'is not found in installation param:%s' %
+                              (CONDITIONAL_PARAMS_REQUIRED,
+                               conditional_params['installationParam']))
+            install_params = {}
+            install_params['antennaAzimuth'] = conditional_params['installationParam'][
+                'antennaAzimuth']
+            install_params['longitude'] = conditional_params['installationParam'][
+                'longitude']
+            install_params['latitude'] = conditional_params['installationParam'][
+                'latitude']
+            install_params['antennaGain'] = conditional_params['installationParam'][
+                'antennaGain']
+            install_params['indoorDeployment'] = conditional_params['installationParam'][
+                'indoorDeployment']
+            install_params['antennaBeamwidth'] = conditional_params['installationParam'][
+                'antennaBeamwidth']
+            install_params['height'] = conditional_params['installationParam']['height']
+            device.update({
+                'installationParam': install_params
+            })
+            device.update({
+                'cbsdCategory': conditional_params['cbsdCategory']
+            })
+      else:
+        # Assert the REG-Conditionals for Category A Device required for PPA reference model
+        logging.debug("else")
```

Comment:
Please remove.

Reply:
Debug   log was added for testing purpose has been removed now